pkg/resource/queue/sdk.go

```diff
@@ -57,20 +66,26 @@ func (rm *resourceManager) sdkFind(
 	if err != nil {
 		return nil, err
 	}
-
-	_, respErr := rm.sdkapi.GetQueueAttributesWithContext(ctx, input)
-	rm.metrics.RecordAPICall("GET_ATTRIBUTES", "GetQueueAttributes", respErr)
-	if respErr != nil {
-		if awsErr, ok := ackerr.AWSError(respErr); ok && awsErr.Code() == "UNKNOWN" {
+	var resp *svcsdk.GetQueueAttributesOutput
+	resp, err = rm.sdkapi.GetQueueAttributesWithContext(ctx, input)
+	rm.metrics.RecordAPICall("GET_ATTRIBUTES", "GetQueueAttributes", err)
+	if err != nil {
+		if awsErr, ok := ackerr.AWSError(err); ok && awsErr.Code() == "UNKNOWN" {
 			return nil, ackerr.NotFound
 		}
-		return nil, respErr
+		return nil, err
 	}
 
 	// Merge in the information we read from the API call above to the copy of
 	// the original Kubernetes object we passed to the function
 	ko := r.ko.DeepCopy()
 
+	if ko.Status.ACKResourceMetadata == nil {
+		ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	tmpARN := ackv1alpha1.AWSResourceName(*resp.Attributes["QueueArn"])
+	ko.Status.ACKResourceMetadata.ARN = &tmpARN
+
```

Comment:
As I suspected, the code generator is broken for attribute-based APIs like SQS. I need to fix `ack-generate` to properly set CR fields to the values of attributes returned.